dask/array/overlap.py

```diff
@@ -583,30 +609,121 @@ def map_overlap(x, func, depth, boundary=None, trim=True, **kwargs):
            [16,  17,  18,  19],
            [20,  21,  22,  23],
            [24,  25,  26,  27]])
-    """
-    depth2 = coerce_depth(x.ndim, depth)
-    boundary2 = coerce_boundary(x.ndim, boundary)
 
-    for i in range(x.ndim):
-        if isinstance(depth2[i], tuple) and boundary2[i] != "none":
-            raise NotImplementedError(
-                "Asymmetric overlap is currently only implemented "
-                "for boundary='none', however boundary for dimension "
-                "{} is {}".format(i, boundary2[i])
+    The ``da.map_overlap`` function can also accept multiple arrays.
+
+    >>> func = lambda x, y: x + y
+    >>> x = da.arange(8).reshape(2, 4).rechunk((1, 2))
+    >>> y = da.arange(4).rechunk(2)
+    >>> da.map_overlap(func, x, y, depth=1).compute() # doctest: +NORMALIZE_WHITESPACE
+    array([[0,  2,  4,  6],
+           [4,  6,  8,  10]])
```

Comment:
```suggestion
    array([[ 0,  2,  4,  6],
           [ 4,  6,  8,  10]])
```

numpy likes to pad out the array reprs and `NORMALIZE_WHITESPACE` treats blocks of whitespace the same way, but `[0` and `[ 0` are viewed as different things.  You can try the above to see if it resolves the doctest failures.